dist/templates/k8s.ovn.org_egressfirewalls.yaml.j2

```diff
@@ -82,16 +82,63 @@ spec:
+                          description: nodeSelector will allow/deny traffic to the
+                            Kubernetes node IP of selected nodes. If this is, set
+                            cidrSelector and DNSName must be unset.
+                          properties:
+                            matchExpressions:
+                              description: matchExpressions is a list of label selector
+                                requirements. The requirements are ANDed.
+                              items:
+                                description: A label selector requirement is a selector
+                                  that contains values, a key, and an operator that
+                                  relates the key and values.
+                                properties:
+                                  key:
+                                    description: key is the label key that the selector
+                                      applies to.
+                                    type: string
+                                  operator:
+                                    description: operator represents a key's relationship
+                                      to a set of values. Valid operators are In,
+                                      NotIn, Exists and DoesNotExist.
+                                    type: string
+                                  values:
+                                    description: values is an array of string values.
+                                      If the operator is In or NotIn, the values array
+                                      must be non-empty. If the operator is Exists
+                                      or DoesNotExist, the values array must be empty.
+                                      This array is replaced during a strategic merge
+                                      patch.
+                                    items:
+                                      type: string
+                                    type: array
+                                required:
+                                - key
+                                - operator
+                                type: object
+                              type: array
+                            matchLabels:
+                              additionalProperties:
+                                type: string
+                              description: matchLabels is a map of {key,value} pairs.
+                                A single {key,value} in the matchLabels map is equivalent
+                                to an element of matchExpressions, whose key field
+                                is "key", the operator is "In", and the values array
+                                contains only "value". The requirements are ANDed.
+                              type: object
+                          type: object
                       type: object
-                      minProperties: 1
```

Comment:
I don't understand why we removed min and max properties here? Even if we added a new `nodeSelector`, the rule still holds that we need at least one of them set (nodeselector/dnsname/cidrselector) and that only max one can be set which is clear from the description of each field.
Removing kubebuilder rules breaks backwards compatibility of how API would have complained around this and now it would stop complaining if more than one is set OR none are set.

Reply:
lol I know why this was removed, its because we never had it in the first place! Seems like it was hardcoded:
We might need to change the types.go file to actually reflect and incorporate this:
```
// to is the target that traffic is allowed/denied to
// +kubebuilder:validation:MinProperties:=1
// +kubebuilder:validation:MaxProperties:=1
To EgressFirewallDestination `json:"to"`
```
Reason being API will now start behaving differently than before if we remove these validation hooks.